meson_options.txt

```diff
@@ -418,8 +418,6 @@ option('dbus', type : 'combo', choices : ['auto', 'true', 'false'],
 
 option('gnu-efi', type : 'combo', choices : ['auto', 'true', 'false'],
        description : 'gnu-efi support for sd-boot')
-option('efi-cc', type : 'array',
-       description : 'the compiler to use for EFI modules')
```

Comment:
FWIW judging by https://github.com/systemd/systemd/pull/10220 `efi-cc` is used in the wild so it seems it can break build scripts suddenly. I wonder if it would be better to somehow "deprecate" it instead of just removing it?

Reply:
That one is about (presumably) `CC="ccache cc" meson` failing. It also refers to df7cacae696ae3c1d13d2f9a4db24815e07e29a1, which talks about a proper cross-build. So i tend to disagree on this. Currently, using efi-cc is *very* likely not going to do what one intends to.

Reply:
> using efi-cc is very likely not going to do what one intends to.

I agree it makes sense to fail hard then.

One last thing. It doesn't seem to work when systemd is compiled with compiler wrappers like hfuzz-clang/hfuzz-clang++ but it seems it's much easier to just pass -Dgnu-efi=false than to make them work together.

Reply:
If there's an easy way to detect them that would be best. In the worst case you can always hardcode a denylist for gnu-efi=auto.

Reply:
Does compilation for this test file break on those wrappers when run like this: `$CC -nodefaultlibs -lc test.c`?
```c
int main(int argc, char **argv) {
        return argc;
}
```

Reply:
With hfuzz-gcc/hfuzz-clang it does
```
$ gcc test.c
$ hfuzz-gcc test.c
$ hfuzz-gcc -nodefaultlibs -lc  test.c
/usr/bin/ld: /tmp/libhfuzz.1000.c49a3fb6f63a0022.a(instrument.o): undefined reference to symbol '__tls_get_addr@@GLIBC_2.3'
/usr/bin/ld: /lib64/ld-linux-x86-64.so.2: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
```
With afl-gcc it seems to work
```
$ afl-gcc -nodefaultlibs -lc test.c
afl-cc 2.57b by <lcamtuf@google.com>
afl-as 2.57b by <lcamtuf@google.com>
[+] Instrumented 1 locations (64-bit, non-hardened mode, ratio 100%).
```

Reply:
In https://github.com/systemd/systemd/pull/21859 I just looked for "afl-clang/hfuzz-clang" to detect them and replace with clang. But in that case it was easy because BPF skeletons shouldn't be instrumented and they aren't linked against anything. With "boot" it seems it's not that easy as far as I can tell. At least I haven't figured it out yet